src/components/dialogs/generator-creation-dialog.js

```diff
@@ -606,8 +606,12 @@ const GeneratorCreationDialog = ({
                                     <Grid container spacing={2}>
                                         <Grid item xs={12}>
                                             {reactiveCapabilityCurveErrors.map(
-                                                (messageDescriptorId) => (
+                                                (
+                                                    messageDescriptorId,
+                                                    index
+                                                ) => (
                                                     <div
+                                                        key={index}
```

Comment:
would be better to store a logical unique identifier for the error and use it as a key. For example, field X can only ever have one error of type "NonNegativeFoo" so the key may be "X-NonNegativeFoo"

Reply:
Yes, we could do it, but is it necessary ? The keys are never used, they are only there to avoid a warning.
As per the documentation :

> Keys Must Only Be Unique Among Siblings
> 
> Keys used within arrays should be unique among their siblings. However, they don’t need to be globally unique. We can use the same keys when we produce two different arrays

Source : https://reactjs.org/docs/lists-and-keys.html#keys-must-only-be-unique-among-siblings

Reply:
After talking about it, it is better to add the correct key because, among other reasons, its a better practice and do not cost anything.